hackathon_site/dashboard/frontend/src/components/cart/CartCard/CartCard.tsx

```diff
@@ -122,6 +129,7 @@ export const CartCard = ({ hardware_id, quantity, error }: CartCardProps) => {
                     quantity_remaining={hardware.quantity_remaining}
                     handleChange={handleChange}
                     numInCart={quantity}
+                    constraintMax={hardware.max_per_team}
```

Comment:
The constraintMax actually includes the max_per_team of the hardware and category, we'll want to take the minimum between the hardware and category constraint, also, max_per_team is not a required field in hardware or category so constraintMax could be undefined, consider typing this prop as (number | null) or (number | undefined) depending on how you want to implement the constraint.

For this you'll need to get the constraint from the categories so you'll need to use the categoriesSelectors. You can use the constraint calculation in [ProductOverview.tsx](https://github.com/ieeeuoft/hackathon-template/blob/develop/hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.tsx#L302) for reference

Reply:
Is this Quantity Selector part of the code fine then? Where I use this dropdownNum(below)
`const dropdownNum = !constraintMax
        ? quantity_remaining
        : Math.min(quantity_remaining, constraintMax);`
Do I make the changes to CartCard function?